src/frontend/src/binder/expr/mod.rs

```diff
@@ -118,6 +123,26 @@ impl Binder {
         .into())
     }
 
+    pub(super) fn bind_in_list(
+        &mut self,
+        expr: Expr,
+        list: Vec<Expr>,
+        _negated: bool,
+    ) -> Result<ExprImpl> {
+        // FIXME: Handle `not in`.
+        let mut bound_expr_list = vec![self.bind_expr(expr)?];
+        for elem in list {
+            bound_expr_list.push(self.bind_expr(elem)?);
+        }
+        // Only look the first two. [0] is the value, [1] is the first of target list and all
+        // subsequent elements in target list should have same type.
```

Comment:
I guess we'll also need to do casts...

Reply:
> do asserts instead of writing comments?

Assert [1].type == [2].type == ... [3].type ? Seems not necessary.

> I guess we'll also need to do casts...

For cast yes, I guess PG supports. I need to investiagte how to reuse `CAST` in this context.